Packages/com.unity.inputsystem/InputSystem/InputManager.cs

```diff
@@ -3148,6 +3152,15 @@ private unsafe void OnUpdate(InputUpdateType updateType, ref InputEventBuffer ev
                     }
                 }
 
+                // Give the device a chance to do something with data before we propagate it to event listeners.
```

Comment:
Wonder could we clean this bit up a little, like


```
  if (!skipEvent && device.hasEventPreProcessor && !((IEventPreProcessor)device).PreProcessEvent(currentEventReadPtr))
  {
    // Skip event if PreProcessEvent considers it to be irrelevant.
    m_InputEventStream.Advance(false);
    continue;
  }
```
I'm losing track of all the skipping and the leaving and whatnot. Seems clearer to do it in place.

Reply:
fixed